mindsdb/api/a2a/agent.py

```diff
@@ -118,203 +87,62 @@ def invoke(self, query, session_id) -> Dict[str, Any]:
-
-                    # Extract text from parts
                     text = ""
                     for part in msg_dict.get("parts", []):
                         if part.get("type") == "text":
                             text = part.get("text", "")
                             break
-
                     if text:
                         if role == "user":
                             formatted_messages.append({"question": text, "answer": None})
                         elif role == "assistant" and formatted_messages:
-                            # Add the answer to the last question
                             formatted_messages[-1]["answer"] = text
-
-            # Add the current query to the messages
             formatted_messages.append({"question": query, "answer": None})
-
             logger.debug(f"Formatted messages for agent: {formatted_messages}")
-
-            # Use the streaming_invoke method to get real streaming responses
-            streaming_response = self.streaming_invoke(formatted_messages)
-
-            # Yield all chunks directly from the streaming response
-            for chunk in streaming_response:
-                # Only add required fields if they don't exist
-                # This preserves the original structure as much as possible
-                if "is_task_complete" not in chunk:
-                    chunk["is_task_complete"] = False
-
-                if "metadata" not in chunk:
-                    chunk["metadata"] = {}
-
-                # Ensure parts exist, but try to preserve original content
-                if "parts" not in chunk:
-                    # If content exists, create a part from it
-                    if "content" in chunk:
-                        chunk["parts"] = [{"type": "text", "text": chunk["content"]}]
-                    # If output exists, create a part from it
-                    elif "output" in chunk:
-                        chunk["parts"] = [{"type": "text", "text": chunk["output"]}]
-                    # If actions exist, create empty parts
-                    elif "actions" in chunk or "steps" in chunk or "messages" in chunk:
-                        # These chunks have their own format, just add empty parts
-                        chunk["parts"] = []
-                    else:
-                        # Skip chunks with no content
-                        continue
-
-                yield chunk
-
+            streaming_response = self.streaming_invoke(formatted_messages, timeout=timeout)
+            async for chunk in streaming_response:
+                # Only wrap once, and extract content cleanly
+                content_value = chunk.get("text") or chunk.get("output") or json.dumps(chunk)
+                wrapped_chunk = {"is_task_complete": False, "content": content_value, "metadata": {}}
+                if "parts" not in wrapped_chunk:
+                    if "content" in wrapped_chunk:
+                        pass  # keep as is
```

Comment:
looks like this block does nothing and this will work the same without it
```python
           if "parts" not in wrapped_chunk:
                    if "content" in wrapped_chunk:
                        pass  # keep as is
```